### PR TITLE
Fix: <enum>::values() is a static method, was not static

### PR DESCRIPTION
**Pull request by Kimmmey**
Hi everyone,

while working with enums, I noticed that javaparser recognices <enum>::values() as a not static method.

This pull request fixes this bug and adds an additional test.

Best regards
Kim

### CODE REVIEW

File: javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/declarations/JavaParserEnumDeclaration.java [hunk 1]
```diff
@@ -387,7 +387,7 @@ public boolean isDefaultMethod() {
 
         @Override
         public boolean isStatic() {
-            return false;
+            return true;
```
Comment (jlerbsc) [inline javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/declarations/JavaParserEnumDeclaration.java:390]: Can you identify the corresponding section in the JLS?

**Comment by Kimmmey** [comment]
Hi,

the section is jls-8.9.3-100-D.

see: 
Java 8: https://docs.oracle.com/javase/specs/jls/se8/html/jls-8.html#jls-8.9.3-100-D
Java 22: https://docs.oracle.com/javase/specs/jls/se22/html/jls-8.html#jls-8.9.3-100-D

File: javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/declarations/JavaParserEnumDeclaration.java (hunk 1 above)
Comment (Kimmmey) [reply javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/declarations/JavaParserEnumDeclaration.java:390]: Hi,

the section is jls-8.9.3-100-D.

see:
Java 8: https://docs.oracle.com/javase/specs/jls/se8/html/jls-8.html#jls-8.9.3-100-D
Java 22: https://docs.oracle.com/javase/specs/jls/se22/html/jls-8.html#jls-8.9.3-100-D

**Comment by jlerbsc** [comment]
I can't see anything to indicate that all the methods are static, even though that's what your modification is about. I haven't done a more precise test but I'm not sure that Javaparser supports implicit methods.

**Comment by jlerbsc** [comment]
Your PR seems correct. Sorry, I didn't read it properly.

**Pull request merged**